stackslib/src/chainstate/stacks/block.rs

```diff
@@ -567,38 +568,86 @@ impl StacksBlock {
     pub fn validate_transactions_static_epoch(
         txs: &[StacksTransaction],
         epoch_id: StacksEpochId,
+        quiet: bool,
     ) -> bool {
         for tx in txs.iter() {
             if let TransactionPayload::Coinbase(_, ref recipient_opt, ref proof_opt) = &tx.payload {
                 if proof_opt.is_some() && epoch_id < StacksEpochId::Epoch30 {
                     // not supported
-                    error!("Coinbase with VRF proof not supported before Stacks 3.0"; "txid" => %tx.txid());
+                    if !quiet {
+                        error!("Coinbase with VRF proof not supported before Stacks 3.0"; "txid" => %tx.txid());
+                    }
                     return false;
                 }
                 if proof_opt.is_none() && epoch_id >= StacksEpochId::Epoch30 {
                     // not supported
-                    error!("Coinbase with VRF proof is required in Stacks 3.0 and later"; "txid" => %tx.txid());
+                    if !quiet {
+                        error!("Coinbase with VRF proof is required in Stacks 3.0 and later"; "txid" => %tx.txid());
+                    }
                     return false;
                 }
                 if recipient_opt.is_some() && epoch_id < StacksEpochId::Epoch21 {
                     // not supported
-                    error!("Coinbase pay-to-alt-recipient not supported before Stacks 2.1"; "txid" => %tx.txid());
+                    if !quiet {
+                        error!("Coinbase pay-to-alt-recipient not supported before Stacks 2.1"; "txid" => %tx.txid());
+                    }
                     return false;
                 }
             }
             if let TransactionPayload::SmartContract(_, ref version_opt) = &tx.payload {
                 if version_opt.is_some() && epoch_id < StacksEpochId::Epoch21 {
                     // not supported
-                    error!("Versioned smart contracts not supported before Stacks 2.1");
+                    if !quiet {
+                        error!("Versioned smart contracts not supported before Stacks 2.1");
+                    }
                     return false;
                 }
             }
             if let TransactionPayload::TenureChange(..) = &tx.payload {
                 if epoch_id < StacksEpochId::Epoch30 {
-                    error!("TenureChange transaction not supported before Stacks 3.0"; "txid" => %tx.txid());
+                    if !quiet {
+                        error!("TenureChange transaction not supported before Stacks 3.0"; "txid" => %tx.txid());
+                    }
                     return false;
                 }
             }
+            match &tx.auth {
+                TransactionAuth::Sponsored(ref origin, ref sponsor) => {
```

Comment:
There might be an opportunity to DRY this up a bit.  You could implement a "supported in epoch?" check for `TransactionSpendingCondition` itself, and then call that here.

Reply:
tx.auth.is_supported_in_epoch @jcnelson added to TransactionAuth instead of TransactionSpendingCondition to move the whole condition there, does it look good now?